src/network/providers/ethereum-provider.ts

```diff
@@ -0,0 +1,214 @@
+import {ethers} from 'ethers';
```

Comment:
This file will eventually replace `./ethereum.js`.
Keeping both, while a bit untidy, allow us to make a gradual migration.